src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -776,12 +799,17 @@ int ompl::geometric::RRTstar::pruneTree(const base::Cost &pruneTreeCost)
         // Iteratively check the two lists until there is nothing to to remove
         while (leavesToPrune.empty() == false)
         {
-            // First empty the leave-to-prune
+            // First empty the current leaves-to-prune
             while (leavesToPrune.empty() == false)
             {
                 // If this leaf is a goal, remove it from the goal set
                 if (leavesToPrune.front()->inGoal == true)
                 {
+                    // Assert that it's not the _best_ goal
```

Comment:
You are not really asserting that. You are still pruning the best goal. Is that really what you want?